packages-exp/app-check-exp/src/internal-api.ts

```diff
@@ -0,0 +1,293 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseApp } from '@firebase/app-exp';
+import {
+  AppCheckTokenResult,
+  AppCheckTokenListener,
+  AppCheckTokenInternal
+} from './types';
+import { getDebugState, getState, setState } from './state';
+import { TOKEN_REFRESH_TIME } from './constants';
+import { Refresher } from './proactive-refresh';
+import { ensureActivated } from './util';
+import { exchangeToken, getExchangeDebugTokenRequest } from './client';
+import { writeTokenToStorage, readTokenFromStorage } from './storage';
+import { getDebugToken, isDebugMode } from './debug';
+import { base64 } from '@firebase/util';
+import { logger } from './logger';
+import { Provider } from '@firebase/component';
+
+// Initial hardcoded value agreed upon across platforms for initial launch.
+// Format left open for possible dynamic error values and other fields in the future.
+export const defaultTokenErrorData = { error: 'UNKNOWN_ERROR' };
+
+/**
+ * Stringify and base64 encode token error data.
+ *
+ * @param tokenError Error data, currently hardcoded.
+ */
+export function formatDummyToken(
+  tokenErrorData: Record<string, string>
+): string {
+  return base64.encodeString(
+    JSON.stringify(tokenErrorData),
+    /* webSafe= */ false
+  );
+}
+
+/**
+ * This function will always resolve.
```

Comment:
Is this a public-facing comment?

We should use present tense instead of "will" if so.  It's a good habit to get into.

Reply:
No, nothing in this file should be public. Fixed though.

Reply:
That's great, because if we write a decent amount of internal docs/comments, habits will form!  Thanks.